modules/solace/solace.go

```diff
@@ -0,0 +1,291 @@
+			cliScript += "no shutdown\n"
+			cliScript += "exit\n"
+
+			// Add topic subscriptions to the queue
+			for _, topic := range topics {
+				cliScript += fmt.Sprintf("queue %s\n", queue)
+				cliScript += fmt.Sprintf("subscription topic %s\n", topic)
+				cliScript += "exit\n"
+			}
+		}
+		cliScript += "exit\nexit\n"
+	}
+
+	req := testcontainers.ContainerRequest{
+		Image:        s.Image,
+		ExposedPorts: ports,
+		Env:          env,
+		Cmd:          nil,
+		WaitingFor:   wait.ForHTTP("/").WithPort("8080/tcp").WithStartupTimeout(1 * time.Minute),
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.ShmSize = shmSize
+		},
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		return err
+	}
+
+	s.Container = container // assign before using s.Container
+
+	// Wait for Solace broker to be fully active before configuring
+	if err := s.waitForSolaceActive(ctx); err != nil {
+		return fmt.Errorf("failed waiting for Solace broker to become active: %w", err)
+	}
+
+	// Copy and execute CLI script inside the container if it was generated
+	if cliScript != "" {
+		// Write the CLI script to a temp file
+		tmpFile, err := os.CreateTemp("", "solace-queue-setup-*.cli")
+		if err != nil {
+			return fmt.Errorf("failed to create temp CLI script: %w", err)
+		}
+		defer os.Remove(tmpFile.Name())
+		if _, err := tmpFile.Write([]byte(cliScript)); err != nil {
+			return fmt.Errorf("failed to write CLI script: %w", err)
+		}
+		if err := tmpFile.Close(); err != nil {
+			return fmt.Errorf("failed to close CLI script: %w", err)
+		}
+
+		// Copy the script into the container at the correct location
+		err = s.Container.CopyFileToContainer(ctx, tmpFile.Name(), "/usr/sw/jail/cliscripts/script.cli", 0o644)
+		if err != nil {
+			return fmt.Errorf("failed to copy CLI script to container: %w", err)
+		}
+
+		// Execute the script
+		code, out, err := s.Container.Exec(ctx, []string{"/usr/sw/loads/currentload/bin/cli", "-A", "-es", "script.cli"})
+		output := ""
+		if out != nil {
+			bytes, readErr := io.ReadAll(out)
+			if readErr == nil {
+				output = string(bytes)
+			} else {
+				output = fmt.Sprintf("[ERROR reading CLI output: %v]", readErr)
+			}
+		}
+		if err != nil {
+			return fmt.Errorf("failed to execute CLI script for queue/topic setup: %w", err)
+		}
+		if code != 0 {
+			return fmt.Errorf("CLI script execution failed with exit code %d: %s", code, output)
+		}
+	}
+
+	return nil
+}
+
+// BrokerURLFor returns the origin URL for a given service
+func (s *SolaceContainer) BrokerURLFor(service Service) (string, error) {
+	host, err := s.Container.Host(s.ctx)
+	if err != nil {
+		return "", err
+	}
+	port, err := s.Container.MappedPort(s.ctx, nat.Port(fmt.Sprintf("%d/tcp", service.Port)))
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s://%s:%s", service.Protocol, host, port.Port()), nil
```

Comment:
issue: use ctr.PortEndpoint instead. See the rest of the modules that calculate an endpoint/URL

Reply:
done.